src/main.rs

```diff
@@ -191,22 +203,37 @@ fn run(matches: clap::ArgMatches) -> Result<(), Error> {
     // We're going to need a source of randomness.
     let mut rng = rand::thread_rng();
 
-    // If requested, choose a random letter then discard all words that do not
-    // begin with that letter.
+    // Handle alliteration, either by eliminating a specified
+    // character, or using a random one.
     if opt_alliterate {
-        // We choose the first letter from the intersection of the first letters
-        // of each word list in `petnames`.
+        // We choose the first letter from the intersection of the
+        // first letters of each word list in `petnames`.
         let firsts =
             common_first_letters(&petnames.adjectives, &[&petnames.adverbs, &petnames.names]);
-        // Choose the first letter at random; fails if there are no letters.
-        match firsts.iter().choose(&mut rng) {
-            Some(c) => petnames.retain(|s| s.starts_with(*c)),
-            None => {
+        // if a specific character was requested for alliteration,
+        // attempt to use it.
+        if let Some(c) = opt_alliterate_char {
+            // alliteration character is validated to not be empty
+            let ch = c.chars().next().unwrap();
+            if firsts.contains(&ch) {
```

Comment:
Now that `c` is a `char` we can simplify this:

```suggestion
            if firsts.contains(&c) {
```